### PR TITLE
[processor/k8sattributes] Parse IP out of net.Addr to correctly tag k8s.pod.ip 

### PR DESCRIPTION
**Pull request by backjo**
Signed-off-by: Jonah Back <jonah@jonahback.com>

**Description:** <Describe what has changed.>
This PR changes the handling of IP tagging in the Kubernetes Attribute Processor to correctly parse the IP of the network connection. Currently, the string used looks like "10.17.70.57:51372" - which includes the port, which is not correct and causes additional metadata about the Kubernetes environment to not be tagged correctly. 

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: processor/k8sattributesprocessor/pod_association.go
Comment (codeboten) [inline processor/k8sattributesprocessor/pod_association.go:94]: Are the cases here covered by unit tests?

**Review by codeboten — COMMENTED** [review]
Thanks for the contribution! Please add an entry to the changelog.

**Comment by bogdandrutu** [comment]
@dmitryax @pmm-sumo please review

File: processor/k8sattributesprocessor/pod_association.go
Comment (backjo) [reply processor/k8sattributesprocessor/pod_association.go:94]: The existing case, net.IPAddr, is covered. I've added new tests for net.TCPAddr and net.UDPAddr

File: processor/k8sattributesprocessor/processor_test.go
Comment (pmm-sumo) [inline processor/k8sattributesprocessor/processor_test.go:355]: I think that iterating the test over TCPAddr, UDPAddr and IPAddr could be somewhat helpful to make sure only the IP address part is extracted in each case

**Review by pmm-sumo — APPROVED** [review]

File: processor/k8sattributesprocessor/processor_test.go
Comment (backjo) [reply processor/k8sattributesprocessor/processor_test.go]: agreed - i simplified this to iterate over a slice of net.Addr

File: CHANGELOG.md
Comment (dmitryax) [inline CHANGELOG.md:43]: nit:
```suggestion

```

File: processor/k8sattributesprocessor/pod_association.go
Comment (dmitryax) [inline processor/k8sattributesprocessor/pod_association.go:100]: Can we collapse all the cases?
```
case *net.UDPAddr, *net.TCPAddr, *net.IPAddr:
```

Comment (backjo) [reply processor/k8sattributesprocessor/pod_association.go:100]: We can't - the Go type switch would keep `addr` as a `net.Addr` type in this case, which does not have the IP property on it.

Comment (dmitryax) [reply processor/k8sattributesprocessor/pod_association.go:100]: Got it. thanks for explanation

**Review by dmitryax — APPROVED** [review]

File: CHANGELOG.md
Comment (dmitryax) [inline CHANGELOG.md:42]:
```suggestion
- `k8sattributeprocessor`: Parse IP out of net.Addr to correctly tag k8s.pod.ip (#7077)

```

**Comment by jpkrohling** [comment]
There's still a linting failure:

```
processor_test.go:303: File is not `gofmt`-ed with `-s` (gofmt)
				IP: net.IPv4(1, 1, 1, 1),
```

**Comment by backjo** [comment]
Fixed the linting errors - can I get a maintainer approval to re-run the workflows?

**Pull request merged**
